code/src/containers/forms/delivery-form/DeliveryForm.tsx

```diff
@@ -0,0 +1,121 @@
+import useCreateOrder from "@/hooks/use-create-order/useCreateOrder";
+import useGetUserDetails from "@/hooks/use-get-user-details/useGetUserDetails";
+import { PostAddress } from "@/types/delivery.types";
+import { PostAddressValidationScheme } from "@/utils/validators/deliverySheme";
+
+import "@/containers/forms/delivery-form/DeliveryForm.scss";
+
+type DeliveryFormProps = {
+  totalPrice: number;
+};
+
+const DeliveryForm = ({ totalPrice }: DeliveryFormProps) => {
+  const { id, firstName, lastName, email } = useGetUserDetails();
+
+  const {
+    handleSubmit,
+    register,
+    control,
+    formState: { errors, isValid }
+  } = useForm<PostAddress>({
+    resolver: zodResolver(PostAddressValidationScheme),
+    defaultValues: {
+      deliveryMethod: deliveryMethods.NOVA_POST.value
+    }
+  });
+
+  const [createOrder, { isLoading }] = useCreateOrder();
+
+  const onSubmit = (postAddress: PostAddress) => {
+    createOrder({
+      userId: id,
+      firstName,
+      lastName,
+      email,
+      ...postAddress
+    });
+  };
+
+  const deliveryMethodItems = Object.entries(deliveryMethods).map(
+    ([key, { translationKey, value, image }]) => (
+      <AppMenuItem key={key} value={value} className="delivery-form__body-item">
+        <AppBox
+          component="img"
+          src={image}
+          alt="delivery method"
+          className="delivery-form__method-image"
+        />
+        <AppTypography translationKey={translationKey} />
+      </AppMenuItem>
+    )
+  );
+
+  return (
+    <AppBox
+      component="form"
+      onSubmit={handleSubmit(onSubmit)}
+      className="delivery-form"
+    >
+      <AppBox className="delivery-form__body">
+        <AppTypography
+          variant="h3"
+          translationKey="deliveryForm.title"
+          className="delivery-form__body-title"
+        />
+        <AppInput
+          {...register("city")}
+          error={Boolean(errors.city)}
+          data-cy="delivery-city"
+          labelTranslationKey="deliveryForm.city"
+          className="delivery-form__body-input"
+        />
+        <AppInput
+          {...register("department")}
+          error={Boolean(errors.department)}
+          labelTranslationKey="deliveryForm.department"
+          data-cy="delivery-department"
+          className="delivery-form__body-input"
+        />
+        <Controller
+          name="deliveryMethod"
+          control={control}
+          render={({ field }) => (
+            <AppSelect
+              {...field}
+              labelId="delivery-method"
+              label="deliveryForm.postMethod"
+              error={Boolean(errors.deliveryMethod)}
+              data-cy="delivery-method"
+              className="delivery-form__method-select"
+              inputProps={{
+                className: "delivery-form__method-select-input"
+              }}
+            >
+              {deliveryMethodItems}
+            </AppSelect>
+          )}
+        />
```

Comment:
Why we use `Controller` here and not just `Select`?

Reply:
removed